### DISCUSSION TITLE
Updating from 2.3.280 to 2.4 failed with "Volume group nsm is exported"

### DISCUSSION THREAD

**Discussion by TheAOWizard** [Security-Onion-Solutions/securityonion#12046] [Unsupported Versions] [answered]
* **Security Onion version as seen in the lower left corner of SOC and in /etc/soversion. For example: 2.4.10**
  - 2.3.280

* **Is this a cloud deployment or on-prem? If on-prem, do you have Internet access or this an airgap installation?**
  - on-prem deployment with internet access

* **Did you install from our Security Onion ISO image or did you perform a network installation?**
  - Installed from Security Onion ISO
 
* **How many nodes do you have?**
  - 11 Nodes

* **What are the hardware specs of each of those nodes?**
  - Manager Node: VM, 8 cores (no MHz limit), 32GB Memory, 1 x OS Disk 100GB, 1 x NSM Disk 500GB, 1 x NIC.
  - Search Nodes: VM, 4 Cores (no MHz limit), 64GB Memory, 1 x OS Disk 100GB, 1 x Storage disk 2TB, 1 x NIC.
  - Receiver Nodes: VM, 4 Cores (no MHz limit), 8GB Memory, 1 x OS Disk 100GB, 1 x NIC.
  - Forward Nodes: Physical Nodes, 40 Cores, 600GB Memory, 1 x OS Disk 600GB, 1 x NSM Disk 6.6TB, 4 x NIC.

* **How are each of those nodes configured? (ex. manager with 2 search nodes and 3 forward nodes)**
  - 1 Manager Node
  - 5 Search Nodes
  - 2 Receiver Nodes
  - 3 Forward Nodes

* **Does so-status show all services running?**
  - Yes

* **Do you get any failures when you run sudo salt-call state.highstate?**
  - No

* **Does the SOC Grid page show any failures?**
  - No

* **Explain your issue. For example: Installation fails when I select this series of options...**
  - I've been following [the 2.4 for migrating appendix](https://docs.securityonion.net/en/2.4/appendix.html)
  - soup is 100% up to date
  - There's up-to-date nsm backups
  - Disabled services successful
  - Restarted the machine and stopped docker containers.
  - Unmounted, volumegroup changed and exported successfully.
  - Booted the machine from the SO ISO.
  - Selected sda as OS disk and sdb as NSM disk.
  - Said yes to format.
  - Gets error "Volume group nsm is exported"
![Skærmbillede 2023-12-07 130236](https://github.com/Security-Onion-Solutions/securityonion/assets/117448056/9d08ce31-0cbb-47c6-9494-0caf2ad6a78f)

* **Provide applicable logs. If you are having problems right after setup, provide /root/sosetup.log. If you are having problems during soup, provide /root/soup.log. If you are having problems with a specific component, provide that component's logs from /opt/so/log/.**
  - Running the installation from a web console, can't do special hotkeys to view logs, and after failed installation I revert from a snapshot. If you think any specific logs would be helpful I'll try to get them, but I guess this is just me missing something essential.

Is it wrong to assume that this fails because the guide expect me to remove a physical disk (NSM) after exporting it?
I've been trying multiple solutions, for-example compressing NSM and scp it to another machine, but without any luck.
Any tips/tricks would be appreciated a lot! Thanks in advance.

**Comment by Acewiza** [comment]
I'm a little squeamish about doing upgrades across major versions, but what happens if you leave out step 6 of ur issue and let the Onion do it's NSM thing?

**Comment by robbiemarshall** [comment, marked as answer]
It looks like from your comments above you installed on top of your old /nsm volume, which is why you received the error. When it asks where you want your root and nsm mounts, set both to whichever drive was previously root, leaving your previous nsm mount alone.  It will be re-imported and used AFTER the OS install with LVM.

**Reply by TheAOWizard** [reply]
I had a hard time figuring out what to choose, but in the end I ended up saying no to place nsm on another drive and now i finally have 2.4 :D
